### ISSUE TITLE
Can't add old libraries ("SteamApps" instead of "steamapps")

### ISSUE THREAD

**Issue by Tithen-Firion** [ValveSoftware/steam-for-linux#4637]
- Steam client version: compiled on 2016.09.20 18:21:08
- Distribution: Ubuntu 16.04 LTS
- Opted into Steam client beta?: No
- Have you checked for system updates?: Yes

I have some old Steam Library for Linux ~~on my NTFS partition~~ (same happens on Ext4). It was created long time ago and `SteamApps` directory name is CamelCased. When I try to add this library I get **New Steam library must be on a filesystem mounted with execute permissions** error. When I rename `SteamApps` to `steamapps` it works like a charm, detects games and lets me play them without any problem.

**Comment by Elektordi** [comment]
Similar problem here. Haven't used Steam for months on a certain computer. Just started it again and installed appid 413150 ("Stardew Valley") and download was failing at the end with "disk write error".
Steam was trying to install the game in "~/.steam/steam/steamapps/common/Stardew Valley" in properties but folder was named "~/.steam/steam/SteamApps/...". Deleted the game and downloading it again, same problem...
I tried to install appid 40800 ("Super Meat Boy") and it installed in the good directory in the first time.

I did a symlink from "SteamApps" to "steamapps" and then retried the download, it was OK immediately for appid 413150. And I can now see that game is installed in "~/.steam/steam/SteamApps/common/Stardew Valley/" in properties.

And for information, Steam was installed few years ago on this computer.
I have a more recent laptop where I installed Steam few weeks ago, the folder is named "~/.steam/steam/steamapps", and the app 413150 was installing without any problem.

-     Steam client version: Build Sep 20 2016 at 18:21:08
-     Distribution: Ubuntu 14.04.5 LTS
-     Opted into Steam client beta?: No
-     Have you checked for system updates?: Yes
